Header

johnhartmann26 opened the pull request:
1. Added logo as a link to "/" in SVG format
2. Added nav with links to "how-it-works", "about", and "search"
3. Added default search value "javascript"

johnhartmann26 committed:
Added & stylized header

1. Added logo as a link to "/" in SVG format
2. Added nav with links to "how-it-works", "about", and "search"
3. Added default search value "javascript"

jkettmann commented on src/components/Header.js:
```diff
@@ -0,0 +1,15 @@
+  return (
+    <header>
+      <Link className="logo" to="/"><Logo /></Link>
+      <Nav />
```
I wouldn't separate the Header and Nav components. It feels like unnecessary abstraction since they both are very simple.

If you decide to leave them separate I'd suggest to create a `src/components/Header` directory and move the Header and Nav components there. The Nav component will only be used inside the Header component and thus it makes sense to keep them together

jkettmann commented on src/App.css:
```diff
@@ -2,4 +2,34 @@
+
+/* header */
+
+header {
```
Using global styles is not recommended for real-world applications. You will ultimately end up with naming conflicts (think about logo class inside header and footer) or you'll need something old-school like BEM.

Have a look at css-modules or (even better IMO) CSS-in-JS like styled-components. Then you can move the styles next to the Header component. Makes it much cleaner and easier to maintain when the app grows

jkettmann commented on src/components/pages/Home.js:
```diff
@@ -3,7 +3,9 @@ import React from 'react';
   return (
     <div>
-      Home
+      <h1 id="home">Home</h1>
```
It's common practice to separate shared components and pages, so you chose a good approach here. But Usually you would have the pages and components directories on the same level.

For example like this

```
- src
-- components
--- Header
---- Header.js
-- pages
--- Home
---- Home.js
```

jkettmann commented on src/components/Header.js:
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
The layout and Zeplin design don't match. Have a look at [this document](https://www.notion.so/Pixel-perfect-layouts-5b37e3c869cf45a4bf05736d94d75394) for more info

![image](https://user-images.githubusercontent.com/9722792/77822342-fea6e080-70f1-11ea-876f-2afc09aed820.png)

jkettmann reviewed (changes requested):
I left a couple of comments. Two more things

1. The PR title can be simply the title of the ticket (Header). You wouldn't usually include a description of what you did
2. I guess you used npm, right? Can you switch to yarn? I'll probably add an automatic PR deployment to simplify reviews and will use yarn there. Everyone working on the code should use the same since we use different lock files. That can lead to nasty dependency conflicts

johnhartmann26 committed:
Header fixes

1. Fixed navbar order and margin
2. Removed nav component and added directly to header
3. Changed path naviagtion
4. Began using Styled Components

jkettmann commented on src/App.css:
```diff
@@ -2,4 +2,20 @@
 
 * {
     font-family: 'Montserrat', sans-serif;
+}
```
Fyi: you could use styled-component's [createGlobalStyle](https://styled-components.com/docs/api#createglobalstyle) instead of a `.css` file. Just to be more consistent. It's up to you though if you want to keep it this way or use `createGlobalStyle`

jkettmann commented on src/components/Header.js:
```diff
@@ -0,0 +1,31 @@
+import styled from 'styled-components';
+import { ReactComponent as Logo } from './logo.svg';
+
+const HeaderStyled = styled.header`
```
Again I leave it to you, but what a lot of people do is to separate the styled components and the function component. Here you could have a Header.js file and a Header.style.js file that looks like

```
export const HeaderStyles = styled.header...

export const Nav = styled.nav...
```

jkettmann commented on src/App.css:
```diff
@@ -2,4 +2,20 @@
     font-family: 'Montserrat', sans-serif;
+}
+
+.logo {
```
These three classes .logo, .nav-link and .nav-link:nth-child should be moved to Header.js as well

jkettmann commented on src/App.css:
```diff
@@ -2,4 +2,20 @@
+.nav-link {
+  text-decoration: none;
+  color: #636363;
+  font-size: 16px;
```
You can add `margin-left: 26px;` here instead of the nth-child rule below. Making it 26px instead of 25px will make it pixel-perfect btw ;-)

jkettmann commented on src/App.css:
```diff
@@ -2,4 +2,20 @@
+.logo {
+  width: 150px;
+  height: 36px;
+  align-self: center;
```
You can add a `margin-top: -2px` to position the logo pixel-perfect. Logo and links were still a bit off
![image](https://user-images.githubusercontent.com/9722792/77829540-f6b36480-7122-11ea-97ed-9a22dcdf0f29.png)

johnhartmann26 committed:
Header Fixes 2: Electric Boogaloo

1. Abolished app.css
2. Switched fully to styled components
3. More perfect-er pixels

johnhartmann26 committed:
Revert "Header Fixes 2: Electric Boogaloo"

This reverts commit 116c294fd2af3df08bc225b5ca7ebdc772063c42.

johnhartmann26 committed:
Revert "Revert "Header Fixes 2: Electric Boogaloo""

This reverts commit d8eb06e877180efa17a0989ab39d7078493240f8.

johnhartmann26 committed:
Header fixes

Fixed an accidential deletion +
1. Abolished app.css
2. Switched fully to styled components
3. More perfect-er pixels

jkettmann commented on src/components/Header.style.js:
```diff
@@ -0,0 +1,31 @@
+`;
+
+export const NavLinkWrap = styled.div`
+  a {
```
This should be a separate component like `export const NavLink = styled(Link)` IMO

jkettmann reviewed (changes requested):
You achieved perfect pixels! There's only one last comment, sorry :-)

johnhartmann26 committed:
Eliminated useless NavLinkWrap

1. Eliminated useless NavLinkWrap

jkettmann reviewed (approved):
Even better :+1:
